engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/AvroSchemaDeterminer.scala

```diff
@@ -0,0 +1,13 @@
+package pl.touk.nussknacker.engine.avro
+
+import cats.data.Validated
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaRegistryError
+
+trait AvroSchemaDeterminer extends Serializable {
+
+  def determineSchemaUsedInTyping: Validated[SchemaRegistryError, Schema]
+
+  def determineSchemaInRuntime: Validated[SchemaRegistryError, Option[Schema]]
```

Comment:
In all implementation it's either:
- Some(schema used in typing)
- None
Is it possible there will be other options? If not, maybe flag would be enough?
Also, maybe let's add comment why schema in runtime could be different?